.github/workflows/aste_ci.yml

```diff
@@ -61,6 +61,7 @@ jobs:
         working-directory: tests/example
         run: |
           ./run.sh
+          ./run_nn_gradient.sh
```

Comment:
Having such an integration test is great. I guess there will be more of them in future. I would propose we apply a more sustainable structure here, namely a hierarchical `run` structure, so that we have

`./run.sh` (executes all integration tests)
` |-nng ` directory containing the run script for nng
` |- ...` other directories ...

Copying the config is fine, maybe sharing the mesh files is possible (in order to reduce the size of the repo). Maybe the top level `run` script can just provide an environment variable for the mesh files located at a unique place.

Reply:
I restructured the test examples now it looks better and more sustainable